src/test/java/net/minecraftforge/debug/world/RegistryAccessExtensionTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.world;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.RegistryFileCodec;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.Biomes;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.world.RegistryAccessExtension;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.event.server.ServerAboutToStartEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.registries.ForgeRegistries;
+import net.minecraftforge.registries.ForgeRegistryEntry;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+@Mod(RegistryAccessExtensionTest.MODID)
+public class RegistryAccessExtensionTest
+{
+    public static final String MODID = "registry_access_extension_test";
+    public static final Logger LOGGER = LogManager.getLogger(MODID);
+
+    public static final ResourceLocation TEST_A_NAME = new ResourceLocation(MODID, "worldgen/test_a");
+    public static final ResourceLocation TEST_B_NAME = new ResourceLocation(MODID, "worldgen/test_b");
+    public static final RegistryAccessExtension<TestA> TEST_A = new RegistryAccessExtension<>(TEST_A_NAME, TestA.class, TestA.DIRECT_CODEC);
+    public static final RegistryAccessExtension<TestB> TEST_B = new RegistryAccessExtension<>(TEST_B_NAME, TestB.class, TestB.DIRECT_CODEC);
```

Comment:
Another thing I realized is that these are statically initialized due to how you set up your system. However, these should be deferred for creating until their associated registry event. This would mean that the registry and register for the dynamic objects would need to be not part of the actual registered object which is how it should be. However, it's still relatively trivial to get the necessary registry without that since you already have the registry key of the registry and can pull it from `RegistryManager#getRegistry` in the `ACTIVE` registry set.

Reply:
This was intentional so that the extension object can take responsibility over setting things up the resourcekey/register properly. Happy to remove it but it makes it harder to enforce things like the path uniqueness that we discussed or ensure the backing registry is built with the right configuration - I could add some static helper methods for creating keys/registers that handle these requirements for the modder